src/components/TokenRow/TokenRow.tsx

```diff
@@ -22,17 +22,17 @@ const CryptoInput = (props: InputProps) => (
   />
 )
 
-type TokenRowProps = {
-  control: Control
-  fieldName: string
+type TokenRowProps<C> = {
+  control: Control<C>
+  fieldName: Path<C>
```

Comment:
The convention for generics is to use `T` to represent the generic type, though it's not mandatory.
```suggestion
type TokenRowProps<T> = {
  control: Control<T>
  fieldName: Path<T>
```